prow/entrypoint/run_test.go

```diff
@@ -0,0 +1,106 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package entrypoint
+
+import (
+	"io/ioutil"
+	"os"
+	"path"
+	"testing"
+
+	"k8s.io/test-infra/prow/pod-utils/wrapper"
+)
+
+func TestOptions_Run(t *testing.T) {
+	var testCases = []struct {
+		name           string
+		args           []string
+		expectedLog    string
+		expectedMarker string
+		expectedErr    bool
+	}{
+		{
+			name:           "successful command",
+			args:           []string{"sh", "-c", "exit 0"},
+			expectedLog:    "",
+			expectedMarker: "0",
+			expectedErr:    false,
+		},
+		{
+			name:           "successful command with output",
+			args:           []string{"echo", "test"},
+			expectedLog:    "test\n",
+			expectedMarker: "0",
+			expectedErr:    false,
+		},
+		{
+			name:           "unsuccessful command",
+			args:           []string{"sh", "-c", "exit 12"},
+			expectedLog:    "",
+			expectedMarker: "12",
+			expectedErr:    true,
+		},
+		{
+			name:           "unsuccessful command with output",
+			args:           []string{"grep"},
+			expectedLog:    "Usage: grep [OPTION]... PATTERN [FILE]...\nTry 'grep --help' for more information.\n",
```

Comment:
this one still makes me a little bit nervous. relying on a script or something in the repo would be a bit more portable.

Reply:
Yeah, I made sure none of that was GNU-specific but can try something fancier to make it more portable